weaviate/datadog_checks/weaviate/check.py

```diff
@@ -0,0 +1,124 @@
+
+                version_parts = {
+                    'major': major,
+                    'minor': minor,
+                    'patch': patch,
+                }
+                self.set_metadata('version', version_raw, scheme='semver', part_map=version_parts)
+            else:
+                self.log.debug("Invalid Weaviate version format: %s", version)
+        else:
+            self.log.debug("Could not retrieve version metadata from host.")
+
+    def _submit_liveness_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_LIVENESS_ENDPOINT)
+        tags = copy.deepcopy(self.tags)
+        tags.append(f"weaviate_liveness_url:{endpoint}")
+
+        start_time = time.time()
+        response = self.http.get(endpoint)
+        end_time = time.time()
+
+        if response.ok:
+            latency = round_value((end_time - start_time) * 1000, 2)
+            self.service_check('liveness.status', AgentCheck.OK, tags)
+            self.gauge('http.latency_ms', latency, tags=tags)
+        else:
+            self.service_check('liveness.status', AgentCheck.CRITICAL, tags)
+
+    def _submit_node_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_NODE_METRICS_ENDPOINT)
+        response = self.http.get(endpoint)
+        if not response.ok:
+            self.log.debug("Could not retrieve Node metrics. Request returned a: %s", str(response.status_code))
+            return
+
+        data = response.json()
+
+        for node in data.get('nodes', []):
+            tags = copy.deepcopy(self.tags)
+
+            tags.append(f"weaviate_node:{node.get('name', '')}")
+            tags.append(f"weaviate_version:{node.get('version', '')}")
+            tags.append(f"weaviate_githash:{node.get('gitHash', '')}")
+
+            if status := node.get('status'):
+                tags.append(f"weaviate_node_status:{status.lower()}")
+                self.gauge('node.status', NODE_STATUS_VALUES.get(status, NODE_STATUS_VALUES['UNKNOWN']), tags=tags)
+                self.service_check(
+                    'node.status', NODE_STATUS_VALUES.get(status, NODE_STATUS_VALUES['UNKNOWN']), tags=tags
+                )
+
+            if stats := node.get('stats'):
+                self.gauge('node.stats.shards', stats.get('shardCount', 0), tags=tags)
+                self.gauge('node.stats.objects', stats.get('objectCount', 0), tags=tags)
+
+            if shards := node.get('shards'):
+                for shard in shards:
+                    tags.append(f"shard_name:{shard.get('name', '')}")
+                    tags.append(f"class_name:{shard.get('class', '')}")
+                    self.gauge('node.shard.objects', shard.get('objectCount', 0), tags=tags)
+
+    def check(self, instance):
+        if self.api_url:
+            try:
+                self._submit_liveness_metrics()
+                self._submit_version_metadata()
+                self._submit_node_metrics()
+            except Exception as e:
```

Comment:
There's a lot of code in here that this `except` clause is handling, and if `_submit_liveness_metrics()` fails, then the other two functions are skipped. Maybe you can instead have the `try-catch` inside each of these individual calls to specifically handle where the errors may happen.

Reply:
Spoke to @iliakur in regards to this, he wanted me to try to limit the amount of try except catching. We think it's better in the sense that we won't swallow any of the exceptions